server/src/main/java/org/candlepin/controller/ImportResult.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2009 - 2016 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.controller;
+
+import org.candlepin.model.Persisted;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+
+/**
+ * The ImportResult class contains references to the entities which were processed by a
+ * controller-level import operation.
+ *
+ * @param <E>
+ *  The entity class contained by a given import result instance
+ */
+public class ImportResult<E extends Persisted> {
+
+    private final Map<String, E> skippedEntities;
+    private final Map<String, E> createdEntities;
+    private final Map<String, E> updatedEntities;
+    private Map<String, E> importedEntities;
+
+    /**
+     * Instantiates a new ImportResult instance, using the given maps as its result set. Changes
+     * made to the maps after creating an import result will be reflected by this object.
+     *
+     * @param skippedEntities
+     *  A mapping of entities which were skipped during the import
+     *
+     * @param createdEntities
+     *  A mapping of entities which were created during the import
+     *
+     * @param updatedEntities
+     *  A mapping of entities which were updated during the import
+     *
+     * @throws IllegalArgumentException
+     *  if any of the provided maps are null
+     */
+    public ImportResult(Map<String, E> skippedEntities, Map<String, E> createdEntities,
+        Map<String, E> updatedEntities) {
+
+        if (skippedEntities == null) {
+            throw new IllegalArgumentException("skippedEntities is null");
+        }
+
+        if (skippedEntities == null) {
+            throw new IllegalArgumentException("skippedEntities is null");
+        }
+
+        if (skippedEntities == null) {
+            throw new IllegalArgumentException("skippedEntities is null");
+        }
+
+        this.skippedEntities = skippedEntities;
```

Comment:
The first version actually was immutable, but after hitting another bug and reevaluating the usage of the maps, I opted to make it nothing more than a wrapper object around the maps since the import method isn't retaining them for state, and protecting them against modification doesn't (currently) buy us anything. Also, copying these maps unnecessarily is pretty expensive.